File: other/PrimeChecker.scala
import scala.io.StdIn
import scala.util.{Try, Success, Failure}

object PrimeChecker {
  
  /**
   * 素数判定を行う関数
   * @param n 判定したい正の整数
   * @return 素数の場合true、そうでなければfalse
   */
  def isPrime(n: Int): Boolean = {
    if (n < 2) false
    else if (n == 2) true
    else if (n % 2 == 0) false
    else {
      // 3から√nまでの奇数で割り切れるかチェック
      val sqrt = math.sqrt(n).toInt
      (3 to sqrt by 2).forall(n % _ != 0)
    }
  }
  
  /**
   * 数値を処理する関数
   * @param number 処理する数値
   */
  def processNumber(number: Int): Unit = {
    number match {
      case n if n <= 0 =>
        println("エラー: 正の整数を入力してください")
      case n if n > 10000000 =>
        println("エラー: 10,000,000以下の数値を入力してください")
      case n if isPrime(n) =>
        println(s"$n は素数です")
      case n =>
        println(s"$n は素数ではありません")
    }
  }
  
  def main(args: Array[String]): Unit = {
    val numberOpt = if (args.nonEmpty) {
      // コマンドライン引数から取得
      Try(args(0).toInt).toOption
    } else {
      // ユーザー入力から取得
      print("判定したい数値を入力してください: ")
      Try(StdIn.readInt()).toOption
    }
    
    numberOpt match {
      case Some(number) => processNumber(number)
      case None => println("エラー: 有効な整数を入力してください")
    }
  }
}

File: other/prime_checker.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
)

// isPrime 素数判定を行う関数
func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n == 2 {
		return true
	}
	if n%2 == 0 {
		return false
	}

	// 3から√nまでの奇数で割り切れるかチェック
	sqrt := int(math.Sqrt(float64(n)))
	for i := 3; i <= sqrt; i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

// processNumber 数値を処理する関数
func processNumber(number int) {
	if number <= 0 {
		fmt.Println("エラー: 正の整数を入力してください")
		return
	}

	if number > 10000000 {
		fmt.Println("エラー: 10,000,000以下の数値を入力してください")
		return
	}

	if isPrime(number) {
		fmt.Printf("%d は素数です\n", number)
	} else {
		fmt.Printf("%d は素数ではありません\n", number)
	}
}

func main() {
	var number int
	var err error

	if len(os.Args) > 1 {
		// コマンドライン引数から取得
		number, err = strconv.Atoi(os.Args[1])
		if err != nil {
			fmt.Println("エラー: 有効な整数を入力してください")
			os.Exit(1)
		}
	} else {
		// ユーザー入力から取得
		fmt.Print("判定したい数値を入力してください: ")
		_, err = fmt.Scanf("%d", &number)
		if err != nil {
			fmt.Println("エラー: 有効な整数を入力してください")
			os.Exit(1)
		}
	}

	processNumber(number)
}